Sw 102

DaniekFjaes opened the pull request:
![image_2023-11-27_125859353](https://github.com/Trimesteret/AngGUI/assets/79269401/8b1c02ed-c08d-4295-baa3-dc31166818a4)

Still W.I.P and has a few issues:
- Needs some specific formatting for lower resolutions, right now it looks subpar
- Missing the ability to change quantity on basket items
- Basket items are currently just a placeholder, until someone actually makes the basket site
- Needs to give a warning when incorrect information is entered.
- Price isn't calculated because it should (probably?) be fetched from a service

DaniekFjaes committed:
Updated html to look slightly more lik uizard
- for some reason, it looks weird until you press a field
- Still needs basket view, delivery options and a confirm button

DaniekFjaes committed:
Changed position of fields and added radio buttons

Still has an issue, where the page doesn't look right until the  form fields are double clicked

vjohns22 committed:
SW-102 * Added some FormControl to remove errors

Made the browser complain less. For now.

DaniekFjaes committed:
Changed all fields to be required

DaniekFjaes committed:
Changed layout and added basket box

DaniekFjaes committed:
Added box for "go to payment"

DaniekFjaes committed:
Added checkboxes for newsletter and terms of service

Added "gå til betaling" button

DaniekFjaes committed:
Merge branch 'develop' into SW-102

DaniekFjaes committed:
Fixed header

DaniekFjaes committed:
added cards for delivery options

DaniekFjaes committed:
Fixed missing formcontrols

DaniekFjaes committed:
Delivery option pricings added

- needs to be replaced with ngFor, so the html doesn't look like shiet

DaniekFjaes committed:
Replaced radio button html with ngFor

DaniekFjaes committed:
ngFor for formfields

ng lint --fix go brrr

DaniekFjaes committed:
added cards to basket w.i.p

DaniekFjaes committed:
Fixed parts of the basket

- added pictures to basket items
- added scroll, if too many items are in the basket
- added total price and such (placeholders right now)

DaniekFjaes committed:
Merge branch 'develop' into SW-102

DaniekFjaes committed:
very important commit

Thekid2002 commented on src/app/webshop/checkout/checkout.component.ts:
```diff
@@ -7,11 +7,64 @@ import { FormGroup, FormControl } from '@angular/forms';
 })
 export class CheckoutComponent {
   checkoutForm = new FormGroup({
```
Use formbuilder

Thekid2002 commented on src/app/webshop/checkout/checkout.component.ts:
```diff
@@ -7,11 +7,64 @@ import { FormGroup, FormControl } from '@angular/forms';
+    { value: 'dao', cost: 59, label: 'DAO' }
+  ];
+
+  addressFields = [
```
Use wizard service : OR remove 🗡️

Thekid2002 commented on src/app/webshop/checkout/checkout.component.ts:
```diff
@@ -7,11 +7,64 @@ import { FormGroup, FormControl } from '@angular/forms';
+    for (const item of basketContent) {
+      const itemPrice = parseFloat(item.price);
+
+      if (!isNaN(itemPrice)) {
```
Hent pris fra API <3

DaniekFjaes replied:
Jeg får vel itemDTOs fra basket, så giver det mening at hente priserne igen, når jeg allerede (burde) have dem?

Thekid2002 commented on src/app/webshop/checkout/checkout.component.html:
```diff
@@ -1,18 +1,79 @@
-      <input matInput id="address" type="text" formControlName="address">
-    </mat-form-field>
+<app-front-basic-layout>
+  <div fxLayout = "row" fxFlexFill>
```
No space in equal

DaniekFjaes committed:
PR changes

- Removed ngFor for form fields and radio buttons
- Fixed spaces around equals
- Added formbuilder

Thekid2002 commented on src/app/webshop/checkout/checkout.component.ts:
```diff
@@ -1,18 +1,50 @@
+    { value: 'dao', cost: 59, label: 'DAO' }
+  ];
+  // Placeholder. This must be fetched from the actual basket, so a service should probably be made to get that information
+  basketContent = [
```
DEn skal hentes fra orderservice

Thekid2002 commented on src/app/webshop/checkout/checkout.component.ts:
```diff
@@ -1,18 +1,50 @@
+  ];
   createCheckout():void{
-    console.log(this.checkoutForm.value);
+    console.log(this.checkoutForm);
```
We do not console.log in production code (unless errors) <3

Thekid2002 commented on src/app/webshop/checkout/checkout.component.ts:
```diff
@@ -1,18 +1,50 @@
+  ngOnInit() : void {
+    this.checkoutForm = this.fb.group({
+      // Define your form controls here using FormBuilder
+      // Example: 'name': [initialValue, [validators]]
```
Remove comments

Thekid2002 commented on src/app/webshop/checkout/checkout.component.ts:
```diff
@@ -1,18 +1,50 @@
+    { name: 'God Gammeldags Rødvin', price: '300', imageUrl: 'assets/PeanutNoar.jfif' },
+    { name: 'Viktor Special', price: '15', imageUrl: 'assets/PeanutNoar.jfif' }
+  ];
   createCheckout():void{
```
Add spacing and comments + should be public

Thekid2002 commented on src/app/webshop/checkout/checkout.component.ts:
```diff
@@ -1,18 +1,50 @@
-    console.log(this.checkoutForm.value);
+    console.log(this.checkoutForm);
+  }
+  goToPayment():void{
```
Add spacing and comment + should be public

DaniekFjaes committed:
Merge branch 'develop' into SW-102

DaniekFjaes committed:
Review changes

-- Removed duplicate submit function
-- Added spaces around functions
-- Checkout now gets information from basket

DaniekFjaes committed:
Fixed routing to checkout

DaniekFjaes committed:
Fixed total price

DaniekFjaes committed:
Added quantity to basket items

DaniekFjaes committed:
Merge branch 'develop' into SW-102

DaniekFjaes committed:
Beautify html

Thekid2002 committed:
Merge branch 'develop' into SW-102

Thekid2002 committed:
SW-102 * Fixed checkout on webshop

Thekid2002 committed:
SW-102 * Added some needed spacing
